src/utils/url-utils.ts

```diff
@@ -60,28 +60,27 @@ export const useUrlHistory = () => {
   };
 
   const removeLayerFromUrl = (layerKey: UrlLayerKey, layerId: string) => {
-    const removeKey = (): string => {
-      urlParams.delete(layerKey);
-
-      if (layerKey === UrlLayerKey.HAZARD) {
-        urlParams.delete('date');
-      }
-
-      return urlParams.toString();
-    };
-
+    // Get all layer ids from the url.
     const urlLayers = urlParams.get(layerKey);
 
     const selectedLayersUrl = urlLayers !== null ? urlLayers.split(',') : [];
-
     const filteredSelectedLayers = selectedLayersUrl
       .filter(l => l !== layerId)
       .join(',');
 
-    const urlToReplace =
-      filteredSelectedLayers === '' ? removeKey() : filteredSelectedLayers;
+    // If the list of layers is empty, remove the layerKey from the url.
+    if (filteredSelectedLayers === '') {
+      urlParams.delete(layerKey);
 
-    replace({ search: urlToReplace });
+      // For hazard layer, remove also the date.
```

Comment:
Could we simply delete the date every time when the list is empty and a date exists?

Reply:
> there is a possibility that a baselinelayer (admin level data) might have dates. I think there has been previous discussions about including dates in an admin level data layer

correct - https://github.com/WFP-VAM/prism-frontend/pull/555

Reply:
we'll need to revisit the logic once admin level dates are incorporated